### PR TITLE
WC: Playlists page

### PR DESCRIPTION
**Pull request by Prabhav-Sharma**
## Playlists Page
I've added the playlist page, which lists all the user's playlists along and renders the playlist items on the page when the user clicks on any playlist title. I've also added a Horizontal Card component to display playlist videos.

### Other Changes
- Added playlist API calls
- Added add PlaylistModal component
- Shifted Action buttons to their own components for reusability.

### Files to check 
- [ ] src/components/ActionButtons/ActionButton.module.css
- [ ] src/components/ActionButtons/LikesButton.jsx
- [ ] src/components/ActionButtons/PlaylistButton.jsx
- [ ] src/components/ActionButtons/WatchLaterButton.jsx
- [ ] src/components/Card/HorizontalVideoCard/HorizontalVideoCard.jsx
- [ ] src/components/PlaylistModal/PlaylistModal.jsx
- [ ] src/components/PlaylistModal/PlaylistModal.module.css
- [ ] src/pages/Playlists/Playlists.jsx
- [ ] src/pages/Playlists/Playlists.module.css
- [ ] src/services/playlistsAPI.js

P.S: Ignore other files.

### CODE REVIEW

File: src/components/ActionButtons/PlaylistButton.jsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+function PlaylistButton({ video, text = false }) {
+  const {
+    state: { playlists },
+  } = useUserData();
+
+  const { toggle: modalToggle, setToggle: setModalToggle } = useToggle(false);
+
+  const videoInPlaylists = playlists.some((playlist) =>
+    playlist.videos.some((v) => v._id === video._id)
```
Comment (deepak29-git) [inline src/components/ActionButtons/PlaylistButton.jsx]: use generic name instead of v

File: src/components/ActionButtons/WatchLaterButton.jsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import styles from "./ActionButton.module.css";
+import { MdOutlineWatchLater } from "react-icons/md";
+
+const WatchLaterButton = ({ video, text = false }) => {
+  const {
+    state: { watchlater },
+    dispatch: userDataDispatch,
+  } = useUserData();
+
+  const token = localStorage.getItem("token");
```
Comment (VedanthB) [inline src/components/ActionButtons/WatchLaterButton.jsx]: you can do this in the AuthProvider / Auth context file.

get the token and then store it in context , and you can use useAuth to get the token anywhere in the app

**Review by VedanthB — APPROVED** [review]
Good work Prabhav,

things i liked 

- clean organised folder structure
- good variable naming
- use of component lib 

You can merge the the PR 🚀

**Review by ZuberDunge — APPROVED** [review]
Well structured code, everything looks fine, good to go from my side.

File: src/components/index.js [hunk 3]
```diff
@@ -2,5 +2,20 @@ import VideoCard from "./Card/VideoCard/VideoCard";
...
 import TextInput from "./TextInput/TextInput";
 import VideoGrid from "./VideoGrid/VideoGrid";
+import PlaylistModal from "./PlaylistModal/PlaylistModal";
+import HorizontalVideoCard from "./Card/HorizontalVideoCard/HorizontalVideoCard";
+import WatchLaterButton from "./ActionButtons//WatchLaterButton";
+import LikesButton from "./ActionButtons/LikesButton";
+import PlaylistButton from "./ActionButtons/PlaylistButton";
 
-export { VideoCard, Navigation, TextInput, VideoGrid };
+export {
```
Comment (deepak29-git) [inline src/components/index.js:11]: this is good approach to export all files from index.js

**Review by deepak29-git — APPROVED** [review]
liked : your folder structure is good.
 suggest one thing use more generic variable name

File: src/components/ActionButtons/WatchLaterButton.jsx (hunk 2 above)
Comment (Prabhav-Sharma) [reply src/components/ActionButtons/WatchLaterButton.jsx]: Yes, that makes more sense. I've made the changes. Thank you

File: src/components/ActionButtons/PlaylistButton.jsx (hunk 1 above)
Comment (Prabhav-Sharma) [reply src/components/ActionButtons/PlaylistButton.jsx]: Done, I've changed it to item. Thanks

**Pull request merged**